backend/tracim_backend/command/cleanup.py

```diff
@@ -0,0 +1,248 @@
+            dest="anonymous_name",
+            required=False,
+        )
+        return parser
+
+    def take_app_action(self, parsed_args: argparse.Namespace, app_context: AppEnvironment) -> None:
+        self._session = app_context["request"].dbsession
+        self._app_config = app_context["registry"].settings["CFG"]
+
+        if parsed_args.dry_run_mode:
+            print("(!) Running in dry-run mode, no changes will be applied.")
+        if parsed_args.force:
+            print("/!\\ Running in force mode, database created may be broken /!\\.")
+        elif parsed_args.best_effort:
+            print(
+                "(!) Running in best effort mode, will anonymise account instead of deleting them if needed."
+            )
+
+        deleted_user_ids = set()
+        deleted_workspace_ids = set()
+
+        with unprotected_content_revision(self._session) as session:
+            uapi = UserApi(
+                config=self._app_config,
+                session=session,
+                current_user=None,
+                show_deleted=True,
+                show_deactivated=True,
+            )
+            user_list = []  # type: typing.List[User]
+            for login in parsed_args.logins:
+                try:
+                    user = uapi.get_one_by_email(login)
+                    user_list.append(user)
+                except UserDoesNotExist:
+                    print("ERROR: user with email {} does not exist".format(login))
+                    exit(1)
+            print("~~~~~~~~~~")
+            for user in user_list:
+                cleanup_lib = CleanupLib(
+                    session, self._app_config, dry_run_mode=parsed_args.dry_run_mode
+                )
+                deleted_user_ids_result = self._delete_user_database_info(
+                    user,
+                    force=parsed_args.force,
+                    best_effort=parsed_args.best_effort,
+                    cleanup_lib=cleanup_lib,
+                    anonymised_user_display_name=parsed_args.anonymous_name,
+                )
+                deleted_user_ids.add(deleted_user_ids_result.user_id)
+                deleted_workspace_ids.update(deleted_workspace_ids)
+                print("~~~~~~~~~~")
+
+            # INFO - G.M - 2019-12-13 - cleanup agenda at end of process
+            if deleted_workspace_ids:
+                deleted_workspace_ids_str = [
+                    '"{}"'.format(workspace_id) for workspace_id in deleted_workspace_ids
+                ]
+                print("Delete agenda of workspaces {}".format(", ".join(deleted_workspace_ids_str)))
+                for workspace_id in deleted_workspace_ids:
+                    try:
+                        cleanup_lib.delete_workspace_agenda(workspace_id)
+                    except FileNotFoundError:
+                        print(
+                            "Warning: Cannot delete agenda for workspace {}, agenda not found".format(
```

Comment:
It is not a problem, the goal is agenda deleted. If already, it is ok.
```suggestion
                            "Info: Agenda for workspace {} already deleted".format(
```

Reply:
Not sure that's a good idea because we cannot know if agenda is "already deleted" or just wasn't create at all or tracim config is broken linking to incorrect folder for agenda.

Reply:
That's not the responsibility of this script. The expected result of the script is "agenda x do not exist". So if it doesn't exist the goal is achieved.

But this is a log and it is not very important, apply suggest only if you want ^^